### PR TITLE
[typescript-fetch] add interfaces, withInterfaces

### PR DESCRIPTION
**Pull request by larshp**
Implements interfaces for typescript-fetch, closes https://github.com/OpenAPITools/openapi-generator/issues/3478

The style from typescript-axios is followed

### PR checklist

- [X] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [X] If contributing template-only or documentation-only changes which will change sample output, [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) before.
- [X] Run the shell script(s) under `./bin/` (or Windows batch scripts under`.\bin\windows`) to update Petstore samples related to your fix. This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit, and these must match the expectations made by your contribution. You only need to run `./bin/{LANG}-petstore.sh`, `./bin/openapi3/{LANG}-petstore.sh` if updating the code or mustache templates for a language (`{LANG}`) (e.g. php, ruby, python, etc).
- [X] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `4.3.x`, `5.0.x`. Default: `master`.
- [X] Copy the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

@TiFu (2017/07) @taxpon (2017/07) @sebastianhaas (2017/07) @kenisteward (2017/07) @Vrolijkx (2017/09) @macjohnny (2018/01) @nicokoenig (2018/09) @topce (2018/10) @akehir (2019/07) @petejohansonxo (2019/11) @amakhrov (2020/02)

### CODE REVIEW

File: samples/client/petstore/typescript-fetch/builds/with-interfaces/apis/PetApi.ts [hunk 1]
```diff
@@ -60,10 +60,146 @@ export interface UploadFileRequest {
     file?: Blob;
 }
 
+/**
+ * PetApi - interface
+ * @export
+ * @interface PetApi
```
Comment (amakhrov) [inline samples/client/petstore/typescript-fetch/builds/with-interfaces/apis/PetApi.ts]: Shouldn't it be `PetApiInterface` instead?
Personally, I do not find this useful at all, as it duplicates the actual definition below. In my code I would rather get rid of that. But if it's here - probably should match the interface name?

Comment (larshp) [reply samples/client/petstore/typescript-fetch/builds/with-interfaces/apis/PetApi.ts]: thanks, fixed

**Comment by larshp** [comment]
thanks for the fast review, looking forward to the 4.3.0 release 😄

**Comment by wing328** [comment]
@larshp thanks for the PR, which has been included in the 4.3.0 release: https://twitter.com/oas_generator/status/1243455743937789952

**Pull request merged**